lexos/static/js/scripts_scrub.js

```diff
@@ -1,267 +1,309 @@
-    // swfileselect, lemfileselect, consfileselect, scfileselect
-    var filetype = $(this).attr('id').replace('bttnlabel', '')
-    usingCache = $('#usecache' + filetype).attr('disabled') != 'disabled'
-
-    if ((usingCache) || ($(this).attr('id') != '')) {
-      // $(this).siblings('.scrub-upload').attr('value', '');
-      // Next two lines clear the file input; it's hard to find a cross-browser solution
-      $('#' + filetype).val('')
-      $('#' + filetype).replaceWith($('#' + filetype).clone(true))
-      $('#usecache' + filetype).attr('disabled', 'disabled')
-      $(this).text('')
-    }
+  $('#punctbox').mousedown(puncTBoxFade())
 
-    // Do Ajax
-    $.ajax({
-      type: 'POST',
-      url: '/removeUploadLabels',
-      data: $(this).text().toString(),
-      contentType: 'text/plain',
-      headers: { 'option': filetype + '[]' },
-      beforeSend: function () {
-        // alert('Sending...');
-      },
-      success: function (response) {
-        // console.log(response);
-      },
-      error: function (jqXHR, textStatus, errorThrown) {
-        console.log('Error: ' + errorThrown)
-      }
-    })
-  })
+  $('#xml-modal').on('show.bs.modal', xmlModalAjax())
 
-  $('#whitespacebox').click(function () {
-    var timeToToggle = 100
-    if ($(this).is(':checked')) {
-      $('#whitespace').removeClass('hidden')
-      // $("#whitespace").fadeIn(timeToToggle);
-    } else {
-      $('#whitespace').addClass('hidden')
-      // $("#whitespace").fadeOut(timeToToggle);
-    }
-  })
-  $('#entityrules').change(function () {
-    console.log($('#entityrules')[0].value)
-    if ($('#entityrules')[0].value == 'MUFI-3' || $('#entityrules')[0].value == 'MUFI-4') {
-      document.getElementById('MUFI-warning').style.display = 'inline-block'
-      $('head').append("<link href='../static/lib/junicode/Junicode.woff' rel='stylesheet' type='text/css'>")
-      $('.filecontents').addClass('Junicode')
-    } else {
-      $('.filecontents').removeClass('Junicode')
-      document.getElementById('MUFI-warning').style.display = 'none'
-    }
-  })
+  $('#xml-modal').on('hidden.bs.modal', removeEmptyTagTable())
+})
 
-  $('#tagbox').click(function () {
-    var timeToToggle = 100
-    if ($(this).is(':checked')) {
-      $('#tag').removeClass('hidden')
-      // $("#whitespace").fadeIn(timeToToggle);
-    } else {
-      $('#tag').addClass('hidden')
-      // $("#whitespace").fadeOut(timeToToggle);
-    }
-  })
+/**
+ * Download the scrubbing results
+ * @returns {void} - returns nothing
+ */
+function downloadScrubbing () {
+  // Unfortunately, you can't trigger a download with an ajax request; calling a
+  // Flask route seems to be the easiest method.
+  window.location = '/downloadScrubbing'
+}
+
+/**
+ * Send ajax request to do scrubbing
+ * @param {string} action - preview or apply
+ * @returns {void} - returns nothing
+ */
+function doScrubbing (action) {
+  if ($('#num_active_files').val() === '0') {
+    $('#error-modal').modal()
```

Comment:
Probably add an error message?

Reply:
What do you mean? There is already an error message in error-modal, this allows it to display.

Reply:
You need to explicitly add an error message here, in case the message is changed in the future. In fact, in just under 20 lines below this line of code, there is a line of code that modifies the error message.